SGL-1 create customized errors.Wrap so can get a single stack trace

nene209 opened the pull request:
- Also allows adding messages as the errors are processed up the chain
 - Update the sql logic to use the new extended errors.Wrap

sagramont reviewed (approved):
We won't be using NewCError now then, we can check if it should log on the error presenter instead

nene209 commented:
Yea - I replaced the call to it as well and updated the error handler logic like this:
`		gqlerr := graphql.DefaultErrorPresenter(ctx, e)
		gqlerr.Message = e.Error()

		if cerr := gle.AsExtendedError(e); cerr != nil {
			log.Error().Msgf("%+v", cerr.Error())
			gqlerr.Message = cerr.UserMsg
		}
		return gqlerr`
